tsdb/engine/tsm1/engine.go

```diff
@@ -124,6 +129,9 @@ func (e *DevEngine) Close() error {
 	e.mu.Lock()
 	defer e.mu.Unlock()
 
+	close(e.done)
```

Comment:
should go before acquisition of the lock, otherwise the goroutines could deadlock